### DISCUSSION TITLE
Parse SOAP Response Envelope

### DISCUSSION THREAD

**Discussion by jtyme** [nextgenhealthcare/connect#5851] [Q&A] [answered]
I am trying to get the body of a Web Sender SOAP Response. When I try to parse the BODY XML i just get the fill response message.

Code:
 `var responseMessage = new XML(response.getMessage());
var responseBody = responseMessage.*::['Body'].toString();

logger.debug('BODY : ' + responseBody.toString());

for each(seg in responseMessage.children()){
	logger.debug('segment name : ' + seg.name());
}`

Here is the response message: 
`<env:Envelope xmlns:env="http://www.w3.org/2003/05/soap-envelope">
    <env:Header/>
    <env:Body xmlns:core="http://www.caqh.org/SOAP/WSDL/CORERule2.2.0.xsd">
        <core:COREEnvelopeRealTimeResponse>
            <PayloadType>X12_271_Response_005010X279A1</PayloadType>
            <ProcessingMode>RealTime</ProcessingMode>
            <PayloadID>RETURNED GUID</PayloadID>
            <TimeStamp>2023-07-17T08:24:08-06:00</TimeStamp>
            <SenderID>2000000000</SenderID>
            <ReceiverID>1111111111</ReceiverID>
            <CORERuleVersion>2.2.0</CORERuleVersion>
            <Payload>    ****271 Message *******</Payload>
            <ErrorCode>Success</ErrorCode>
            <ErrorMessage/>
        </core:COREEnvelopeRealTimeResponse>
    </env:Body>
</env:Envelope>`

The name of the segments in the for each loop are: 
`http://www.w3.org/2003/05/soap-envelope::Body`

If I try `var responseBody = responseMessage['(http://www.w3.org/2003/05/soap-envelope::Body)'];`

I get a null value in responseBody.toString(); 

I am pretty sure I need to escape the http: address but trying http:////www.w3.org//2003//05/soap-envelope didnt seem to change anything. I am pretty sure I am missing something simple, just not sure what.

**Comment by rjkroll** [comment]
In your Summary - Day Types section, for your Destination, under Response, is 'Strip Namespaces' checked or unchecked?  If its checked, uncheck it.

Also, this link might point you in the right direction : https://forums.mirthproject.io/forum/mirth-connect/support/8443-how-do-you-parse-a-web-service-response

**Comment by tonygermano** [comment, marked as answer]
I don't see any issues with your code. If I do
```js
var responseMessage = new XML('<env:Envelope xmlns:env="http://www.w3.org/2003/05/soap-envelope"> <env:Header/> <env:Body xmlns:core="http://www.caqh.org/SOAP/WSDL/CORERule2.2.0.xsd"> <core:COREEnvelopeRealTimeResponse> <PayloadType>X12_271_Response_005010X279A1</PayloadType> <ProcessingMode>RealTime</ProcessingMode> <PayloadID>RETURNED GUID</PayloadID> <TimeStamp>2023-07-17T08:24:08-06:00</TimeStamp> <SenderID>2000000000</SenderID> <ReceiverID>1111111111</ReceiverID> <CORERuleVersion>2.2.0</CORERuleVersion> <Payload>    ****271 Message *******</Payload> <ErrorCode>Success</ErrorCode> <ErrorMessage/> </core:COREEnvelopeRealTimeResponse> </env:Body> </env:Envelope>')
var responseBody = responseMessage.*::['Body'].toString();
```
Then `responseBody` contains the string
```xml
<env:Body xmlns:env="http://www.w3.org/2003/05/soap-envelope" xmlns:core="http://www.caqh.org/SOAP/WSDL/CORERule2.2.0.xsd"><core:COREEnvelopeRealTimeResponse><PayloadType>X12_271_Response_005010X279A1</PayloadType><ProcessingMode>RealTime</ProcessingMode><PayloadID>RETURNED GUID</PayloadID><TimeStamp>2023-07-17T08:24:08-06:00</TimeStamp><SenderID>2000000000</SenderID><ReceiverID>1111111111</ReceiverID><CORERuleVersion>2.2.0</CORERuleVersion><Payload>    ****271 Message *******</Payload><ErrorCode>Success</ErrorCode><ErrorMessage/></core:COREEnvelopeRealTimeResponse></env:Body>
```

What are you expecting?
